wordpress_connector/models/contracts.py

```diff
@@ -251,75 +251,89 @@ def create_sponsorship_job(self, values, form_data):
+                sponsorship.update(
+                    {
+                        "user_id": ambassador_id,
+                        "origin_id": self.env["recurring.contract.origin"]
+                        .search([("event_id", "=", event_id)], limit=1)
+                        .id,
+                    }
+                )
 
-        # Notify staff
-        sponsor_lang = form_data["lang"][:2]
-        staff_param = "sponsorship_" + sponsor_lang + "_id"
-        staff = self.env["res.config.settings"].sudo().get_param(staff_param)
-        notify_text = (
-            "A new sponsorship was made on the website. Please "
-            "verify all information and validate the sponsorship "
-            "on Odoo: <br/><br/><ul>"
-        ) + web_info
+            # Notify staff
+            sponsor_lang = form_data["lang"][:2]
+            staff_param = "sponsorship_" + sponsor_lang + "_id"
+            staff = self.env["res.config.settings"].sudo().get_param(staff_param)
+            notify_text = (
+                "A new sponsorship was made on the website. Please "
+                "verify all information and validate the sponsorship "
+                "on Odoo: <br/><br/><ul>"
+            ) + web_info
 
-        title = _("New sponsorship from the website")
-        if "writepray" in form_data:
-            title = _("New Write&Pray sponsorship from the website")
-        sponsorship.message_post(
-            body=notify_text,
-            subject=title,
-            partner_ids=[staff],
-            type="comment",
-            subtype="mail.mt_comment",
-            content_subtype="html",
-        )
+            title = _("New sponsorship from the website")
+            if "writepray" in form_data:
+                title = _("New Write&Pray sponsorship from the website")
+            sponsorship.message_post(
+                body=notify_text,
+                subject=title,
+                partner_ids=[staff],
+                type="comment",
+                subtype="mail.mt_comment",
+                content_subtype="html",
+            )
 
-        sponsorship.correspondent_id.set_privacy_statement(origin="new_sponsorship")
-        return sponsorship
+            sponsorship.correspondent_id.set_privacy_statement(origin="new_sponsorship")
+            return sponsorship
+        except BaseException as err:
+            """Log the error and send a mail stating that it failed running"""
+            _logger.error("Wordpress create sponsorship job failed", exc_info=True)
+            child.activity_schedule(
+                'mail.mail_activity_data_todo',
+                summary="[URGENT] Wordpress create sponsorship job failed",
+                note="Please verify this new sponsorship made from the website with "
+                     f"following information: {form_data}, "
+                     f"and given following values: {values}, "
+                     f"with following error: {err}",
```

Comment:
Does this work or should we cast to string? Not sure how f-strings work.
```suggestion
                     f"following information: {str(form_data)}, "
                     f"and given following values: {str(values)}, "
                     f"with following error: {str(err)}",
```

Reply:
According to documentation (and simple tests), f-strings work nicely with dicts, but we can cast for safety